### PR TITLE
SAA-590: Fix issue with tariff dates without indeterminate sentence

### PR DESCRIPTION
**Pull request by smcveigh941**
A null exception would have happened if the prisoner had a tariff date but their legal status was something different to Indeterminate sentence

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/util/EarliestReleaseDateHelper.kt [hunk 1]
```diff
@@ -12,13 +12,8 @@ fun determineEarliestReleaseDate(prisoner: Prisoner): EarliestReleaseDate {
...
-    releaseDate = when {
-      prisoner.releaseDate != null -> prisoner.releaseDate
-      hasActualParoleDate(prisoner) -> prisoner.actualParoleDate
-      hasTariffDate(prisoner) -> prisoner.tariffDate
-      else -> null
-    },
-    isTariffDate = prisoner.tariffDate?.isEqual(releaseDate) ?: false,
+    releaseDate = releaseDate,
+    isTariffDate = releaseDate != null && prisoner.tariffDate?.isEqual(releaseDate) ?: false,
     isConvictedUnsentenced = prisoner.legalStatus == Prisoner.LegalStatus.CONVICTED_UNSENTENCED,
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/util/EarliestReleaseDateHelper.kt:17]: Should lines 17-19 also be checking dates are null?  The tests are setting one or some to null also.

**Review by matt-crowson-moj — APPROVED** [review]
LGTM
